kork/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/artifactstore/ArtifactStoreConfiguration.java

```diff
@@ -16,14 +16,22 @@
 package com.netflix.spinnaker.kork.artifacts.artifactstore;
 
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.filters.ApplicationStorageFilter;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.filters.RegexApplicationStorageFilter;
 import com.netflix.spinnaker.kork.artifacts.artifactstore.s3.S3ArtifactStoreConfiguration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Import;
 
 @Configuration
+@ComponentScan
```

Comment:
Please add a comment explaining whywe need this.  Seems like it's to pick up DeserializerHookRegistry and SerializerHookRegistry.  Is that right?  Do we also need `@ComponentScan` in EntityStoreConfiguration?